optuna/samplers/_qmc.py

```diff
@@ -0,0 +1,168 @@
+from collections import OrderedDict
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+import scipy
+
+import optuna
+from optuna import distributions
+from optuna import logging
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = logging.get_logger(__name__)
+
+_SUGGESTED_STATES = (TrialState.COMPLETE, TrialState.PRUNED)
+
+_NUMERICAL_DISTRIBUTIONS = (
+    distributions.UniformDistribution,
+    distributions.LogUniformDistribution,
+    distributions.DiscreteUniformDistribution,
+    distributions.IntUniformDistribution,
+    distributions.IntLogUniformDistribution,
+)
+
+
+class QMCSampler(BaseSampler):
+    def __init__(
+        self,
+        *,
+        seed: Optional[int] = None,
```

Comment:
When `seed` is `None`, no randomization (scrambling) happens in the current QMC sampler. This is to ensure that QMCSampler samples from the same QMC sequences when no arguments are given. This comes in handy when we have multiple `QMCSampler`s running simultaneously. 
However, this is counter-intuitive at the same time because most sampler such as `np.random.RandomState` or `scipy.stats.qmc.Sobol` picks random seed when no seed is given. Now I feel adding the scramble (=randomize) option will clarify this situation better. Do you agree? 
```suggestion
        seed: Optional[int] = None,
        scramble = False, 
```